src/core/flow.js

```diff
@@ -529,6 +529,18 @@ export function Flow({ moves, phases, endIf, turn, events, plugins }) {
     // Reset activePlayers.
     ctx = { ...ctx, activePlayers: null };
 
+    // Remove player from playerOrder
+    playerID = playerID || state.ctx.currentPlayer;
+
+    if (arg && true === arg.remove) {
+      const playOrder = [...ctx.playOrder].filter(i => i != playerID);
+      ctx = { ...ctx, playOrder };
+
+      if (0 === playOrder.length) {
+        return EndPhaseEvent(state);
```

Comment:
Instead of calling `EndPhaseEvent` directly here, you should queue it for processing by adding it to `next`.  See line 377 for an example of how to do this.